### PR TITLE
Symfony 6 support

### PR DESCRIPTION
**Pull request by sidz**
- [x] Add Symfony 6 support
- [x] Mark 4.4 as minimum supported version
- [x] Test adapter against supported versions of symfony packages

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,9 +7,8 @@ PHP=$(shell which php)
 JOBS=$(shell nproc)
 
 # PHP CS Fixer
-PHP_CS_FIXER=vendor/bin/php-cs-fixer
-PHP_CS_FIXER_ARGS=--diff --diff-format=udiff --verbose
-export PHP_CS_FIXER_IGNORE_ENV=1
+PHP_CS_FIXER=./.tools/php-cs-fixer
+PHP_CS_FIXER_URL="https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/v3.2.1/php-cs-fixer.phar"
```
Comment (sidz) [inline Makefile:11]: php-cs-fixer is not compatible with symfony 6 yet.

waiting https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6095 to be merged and new release

File: .php-cs-fixer.php [hunk 2]
```diff
@@ -53,18 +53,17 @@
     ])
     ->ignoreDotFiles(false)
     ->name('*php')
-    ->name('.php_cs.dist')
 ;
 
-return Config::create()
+return (new Config())
```
Comment (sidz) [inline .php-cs-fixer.php:58]: config has been copied from infection/infection repo

File: src/PhpSpecExtraOptions.php [hunk 3]
```diff
@@ -1,50 +0,0 @@
...
- */
-
-declare(strict_types=1);
-
-namespace Infection\TestFramework\PhpSpec;
-
-/**
- * @internal
- */
-final class PhpSpecExtraOptions
```
Comment (sidz) [inline src/PhpSpecExtraOptions.php:41]: looks like dead code. Am I missed smth?

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -13,14 +13,19 @@ on:
...
+      SYMFONY_REQUIRE: ${{ matrix.symfony-require }}
 
     strategy:
       matrix:
-        operating-system: [ubuntu-latest]
         php-version: ['7.4', '8.0', '8.1']
         coverage-driver: [pcov]
+        symfony-require: ['^4', '^5', '^6']
+        exclude:
+          - symfony-require: "^6"
```
Comment (sidz) [inline .github/workflows/ci.yaml:25]: symfony 6 requires `"php": ">=8.0.2"`

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -13,14 +13,19 @@ on:
 jobs:
   tests:
     runs-on: ubuntu-latest
+    env:
+      SYMFONY_REQUIRE: ${{ matrix.symfony-require }}
```
Comment (maks-rafalko) [inline .github/workflows/ci.yaml:17]: is it somehow used by composer to require symfony of particular version during `composer update`? As I don't see where we explicitly require the needed version (`^5`, `^6` ...)

**Review by maks-rafalko — APPROVED** [review]

File: src/PhpSpecExtraOptions.php (hunk 3 above)
Comment (maks-rafalko) [reply src/PhpSpecExtraOptions.php:41]: agree, looks like unused

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (sidz) [reply .github/workflows/ci.yaml:17]: symfony flex does it for us.
https://github.com/symfony/flex/blob/7f45cf796f591ad8c1cc56ccc8edc32d78172258/src/Flex.php#L125

**Comment by sidz** [comment]
merged. Thanks @maks-rafalko

**Pull request merged**